### ISSUE TITLE
Sample results

### ISSUE THREAD

**Issue by deepseek** [aub-mind/arabert#1]
@WissamAntoun @fadybaly 
Hi there,

include some results of the trained model

**Comment by WissamAntoun** [comment]
You mean pretraining loss and NSP accuracy?

**Comment by deepseek** [comment]
No, I mean show examples of input text, and the produced text output by arabert.
- I want to see what are arabert replies.
- Do they make sense or are they gibberish talk.
- Does the words carry the meaning.

SHOW arabert SPEAKING

**Comment by fadybaly** [comment]
> No, I mean show examples of input text, and the produced text output by arabert.
> 
> * I want to see what are arabert replies.
> * Do they make sense or are they gibberish talk.
> * Does the words carry the meaning.
> 
> SHOW arabert SPEAKING

Although AraBERT ''alone'' (and any BERT based model) can be used to generate sentences if desired, the quality of the generated text would be bad, mainly because it's not a bidirectional model. Therefore, its ability to create a coherent sentence is poor.
As you may know, BERT was trained on masked word prediction task and next sentence prediction task. It's not trained to generate text. That being the case, we don't recommend to use AraBERT, at least not by itself, for ''speaking'' purposes.

Please correct me if there has been any successful attempts to use BERT as a text generation model.

**Comment by deepseek** [comment]
Then show samples of it's predictions...

And correct you I will: 
https://arxiv.org/abs/2001.09977
https://ai.googleblog.com/2020/01/towards-conversational-agent-that-can.html

**Comment by fadybaly** [comment]
According to the paper and the blog you shared with us, ''Concretely, Meena has a single Evolved Transformer encoder block and 13 Evolved Transformer decoder blocks''. Basically, this is an encoder and a decoder model with 2.6 billion parameters. Whereas BERT is an encoder model with 110 million parameters.

We will share samples of the masked language prediction task as soon as we can.

**Comment by deepseek** [comment]
Thanks
